sopel/db.py

```diff
@@ -180,9 +185,44 @@ def __init__(self, config):
     def connect(self):
         """Get a direct database connection.
 
-        :rtype: :class:`sqlalchemy.engine.Connection`
+        :return: a proxied DBAPI connection object; see
+                 :meth:`sqlalchemy.engine.Engine.raw_connection()`
+
+        .. important::
+
+           The :attr:`database backend <sopel.config.core_section.db_type>` in
+           use can change how the raw connection object behaves. Unless you're
+           writing (or updating) a plugin that needs to be compatible with
+           Sopel versions older than 7.0, or can get away with supporting only
+           one type of database, you probably want to use :meth:`session` and
+           build (or convert) your plugin to use the SQLAlchemy ORM.
+
+        """
+        if self.type != 'sqlite':
+            # log non-sqlite uses of raw connections for troubleshooting, since
+            # unless the developer had a good reason to use this instead of
+            # `session()`, it indicates the plugin was written before Sopel 7.0
+            # and might not work right when connected to non-sqlite DBs
+            LOGGER.info(
+                "Raw connection requested when db_type is not sqlite\n%s",
```

Comment:
We should add: `Please use 'db.session()' instead to use an SQL Alchemy session`.

Reply:
Rewriting the message rn.